csfieldguide/chapters/content/en/algorithms/sections/whats-the-big-picture.md

```diff
@@ -0,0 +1,148 @@
+
+```python3
+def find_high_score(scores):
+    if len(scores) == 0:
+        print("No high score, table is empty")
+        return -1
+    else:
+        highest_so_far = scores[0]
+        for score in scores[1:]:
+            if score > highest_so_far:
+                highest_so_far = score
+        return highest_so_far
+```
+
+But here's another program that implements exactly the same algorithm, this time in the Scratch language.
+
+{image file-path="img/chapters/highscore-in-scratch.png" alt="High score program in Scratch"}
+
+Both of the above programs are the same algorithm.
+In this chapter we'll look in more detail about what an algorithm is, and why they are such a fundamental idea in computer science.
+Because algorithms exist even if they aren't turned in to programs, we won't need to look at programs at all for this topic, unless you particularly want to.
+
+{glossary-link term="algorithm" reference-text="algorithm cost"}{glossary-link end}
+
+### Algorithm cost
+
+When Computer Scientists are comparing algorithms they often talk about the 'cost' of an algorithm.
+The cost of an algorithm can be interpreted in several different ways, but it is always related to how well an algorithm performs based on the size of its input, *n*.
+In this chapter we will talk about the cost of an algorithm as either the time it takes a program (which performs the algorithm) to complete, or the number of steps that the algorithm makes before it finishes.
+
+For example, one way of expressing the cost of the high score algorithm above would be to observe that for a table of 10 values, it does about 10 sets of operations to find the best score, whereas for a table of 20 scores, it would do about twice as many operations.
+In general the number of operations for a table of *n* items will be proportional to *n*.
+Not all algorithms take double the time for double the input; some take a lot more than double, while others take a lot less.
+That's worth knowing in advance because we usually need our programs to scale up well; in the case of the high scores, if you're running a game that suddenly becomes popular, you want to know in advance that the high score algorithm will be fast enough if you get more scores to check.
+
+{panel type="extra-for-experts"}
+
+# Algorithm complexity
+
+The formal term for working out the cost of an algorithm is [algorithm analysis](https://en.wikipedia.org/wiki/Analysis_of_algorithms), and we often refer to the cost as the algorithm's *complexity*.
+The most common complexity is the "time complexity" (a rough idea of how long it takes to run), but often the "space complexity" is of interest - how much memory or disk space will the algorithm use up when it's running?
+
+There's more about how the cost of an algorithm is described in industry, using a widely agreed on convention called 'Big-O Notation', in the ["The whole story!"](the-whole-story) section at the end of this chapter.
```

Comment:
These changes seem to be putting even more restriction on how links are interpreted. This link breaks now so the page doesn't load

Reply:
The syntax is just incorrect.

Reply:
Sorry, old code that snuck through the merge conflict.